apps/storybook/src/ComboBox.stories.tsx

```diff
@@ -478,6 +501,57 @@ WithCustomMessageIcon.args = {
   ),
 };
 
+export const Loading: Story<Partial<ComboBoxProps<string>>> = (args) => {
+  const [options, setOptions] = React.useState<SelectOption<string>[]>([]);
+  const [selectedValue, setSelectedValue] = React.useState<string>();
+  const [isLoading, setIsLoading] = React.useState(true);
+
+  const onChange = React.useCallback((value: string) => {
+    action(value ?? '')();
+    setSelectedValue(value);
+  }, []);
+
+  const emptyContent = React.useMemo(() => {
+    return isLoading ? (
+      <>
+        {new Array(6).fill(null).map((_, index) => {
+          return (
+            <MenuItemSkeleton
+              key={index}
+              hasIcon
+              hasSublabel
+              contentWidth={Math.min(Math.max(Math.random() * 100, 25), 60)}
+            />
+          );
+        })}
+      </>
+    ) : (
+      'No options found'
+    );
+  }, [isLoading]);
+
+  return (
+    <ComboBox
+      inputProps={{ placeholder: 'Select a country' }}
+      value={selectedValue}
+      onChange={onChange}
+      {...args}
+      emptyStateMessage={emptyContent}
+      options={options}
+      onShow={React.useCallback(async () => {
```

Comment:
Maybe reload data on every opening? Or that would mess up too much...

Reply:
Tried, it is showing errors in console that it can't read `__originalIndex` of undefined